### PR TITLE
Remove pinning migrator

### LINKED ISSUE #283: perl migrator
Many old recipes that use `perl` have explicit `perl` version pinnings (because of an old `conda-build` bug), which causes problems with new recipes that try to use them: e.g. https://github.com/conda-forge/perl-xml-parser-feedstock/pull/1 / https://github.com/conda-forge/staged-recipes/pull/4858#discussion_r203149590. It'd be nice to migrate those to not do that.

(I haven't checked how many there are; maybe there's few enough that they should be done explicitly.)

### PR DESCRIPTION
**Pull request by justcalamari**
closes #283

### CODE REVIEW

File: conda_forge_tick/utils.py [hunk 1]
```diff
@@ -7,6 +7,15 @@
 from conda_build.metadata import parse
 
 
+class UniversalSet(set):
```
Comment (CJ-Wright) [inline conda_forge_tick/utils.py]: Why is this needed? (Not saying it's wrong, I just can't see it yet)

Comment (justcalamari) [reply conda_forge_tick/utils.py]: This is the default value for requirements we remove pinnings on. If no specific packages are provided, just remove the pinnings on everything. Since the filter and migrator use set intersection, this is a nice way to make it a no op.

Comment (scopatz) [reply conda_forge_tick/utils.py]: Also, you should not subclass from set, but rather from collections.abc.Set

**Comment by CJ-Wright** [comment]
Is this ok to go in @justcalamari ?

**Comment by justcalamari** [comment]
Yes, @CJ-Wright

**Comment by CJ-Wright** [comment]
I'll merge first thing tomorrow.

**Comment by CJ-Wright** [comment]
Thank you very much @justcalamari
